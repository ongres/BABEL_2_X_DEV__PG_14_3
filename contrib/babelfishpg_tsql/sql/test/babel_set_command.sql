-- Simple SET
SET lc_messages ='fr_FR.utf8';
show lc_messages;
reset lc_messages;

-- Inside transaction with commit
BEGIN;
	SET lc_messages = 'fr_FR.utf8';
COMMIT;
show lc_messages;
reset lc_messages;

-- Inside transaction with rollback
BEGIN;
	SET lc_messages = 'fr_FR.utf8';
ROLLBACK;
show lc_messages;
reset lc_messages;

-- Inside transaction with rollback to savepoint
BEGIN;
	SET lc_messages = 'en_GB.utf8';
	SAVEPOINT SP1;
	SET lc_messages = 'fr_FR.utf8';
	show lc_messages;
	ROLLBACK TO SAVEPOINT SP1;
	show lc_messages;
ROLLBACK;
show lc_messages;
reset lc_messages;

-- Inside procedure
CREATE PROCEDURE lc_proc()
AS $$
begin
	SET lc_messages ='fr_FR.utf8';
	commit;
end;
$$ LANGUAGE plpgsql;
CALL lc_proc();
show lc_messages;
drop procedure lc_proc();
reset lc_messages;

CREATE PROCEDURE lc_proc()
AS $$
begin
	SET lc_messages ='fr_FR.utf8';
	rollback;
end;
$$ LANGUAGE plpgsql;
CALL lc_proc();
show lc_messages;

-- Cleanup
drop procedure lc_proc();
reset lc_messages;
