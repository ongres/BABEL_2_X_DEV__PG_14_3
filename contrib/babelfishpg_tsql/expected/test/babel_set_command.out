-- Simple SET
SET lc_messages ='fr_FR.utf8';
show lc_messages;
 lc_messages 
-------------
 fr_FR.utf8
(1 row)

reset lc_messages;
-- Inside transaction with commit
BEGIN;
	SET lc_messages = 'fr_FR.utf8';
COMMIT;
show lc_messages;
 lc_messages 
-------------
 fr_FR.utf8
(1 row)

reset lc_messages;
-- Inside transaction with rollback
BEGIN;
	SET lc_messages = 'fr_FR.utf8';
ROLLBACK;
show lc_messages;
 lc_messages 
-------------
 C
(1 row)

reset lc_messages;
-- Inside transaction with rollback to savepoint
BEGIN;
	SET lc_messages = 'en_GB.utf8';
	SAVEPOINT SP1;
	SET lc_messages = 'fr_FR.utf8';
	show lc_messages;
 lc_messages 
-------------
 fr_FR.utf8
(1 row)

	ROLLBACK TO SAVEPOINT SP1;
	show lc_messages;
 lc_messages 
-------------
 en_GB.utf8
(1 row)

ROLLBACK;
show lc_messages;
 lc_messages 
-------------
 C
(1 row)

reset lc_messages;
-- Inside procedure
CREATE PROCEDURE lc_proc()
AS $$
begin
	SET lc_messages ='fr_FR.utf8';
	commit;
end;
$$ LANGUAGE plpgsql;
CALL lc_proc();
show lc_messages;
 lc_messages 
-------------
 fr_FR.utf8
(1 row)

drop procedure lc_proc();
reset lc_messages;
CREATE PROCEDURE lc_proc()
AS $$
begin
	SET lc_messages ='fr_FR.utf8';
	rollback;
end;
$$ LANGUAGE plpgsql;
CALL lc_proc();
show lc_messages;
 lc_messages 
-------------
 C
(1 row)

-- Cleanup
drop procedure lc_proc();
reset lc_messages;
